### ISSUE TITLE
NDS rendering still wrong

### ISSUE THREAD

**Issue by YoshiRulz** [TASEmulators/BizHawk#3772]
`client.bufferheight()` is a constant `384` regardless of screen gap, yet trying to draw on the bottom screen (`EmuCore` surface) with `y` values in `192..<384` draws that distance below the origin, not taking the screen gap into account.

If you were wondering why the bundled Lua script draws on the `Client` surface, this is why. If you were wondering why it reads the current rotation/gap settings only to force a preset I have no idea.

I bet this affects Citra too. If only we'd thought to write a robust multi-screen API instead of hacking stuff into the existing one. (This is where I'd say "I told you so," if I could find a link to that.)

**Comment by CasualPokePlayer** [comment]
NDS screen cap is implemented purely on client side; the video buffer is not altered here. This wouldn't affect Citra as it does its screen madness internally so you see it in buffer width/height.

**Comment by CasualPokePlayer** [comment]
> If you were wondering why the bundled Lua script draws on the `Client` surface, this is why.

It doesn't, it uses the emucore (default) surface

**Comment by CasualPokePlayer** [comment]
Calling this fixed with the recent changes to NDS rendering.

**Comment by YoshiRulz** [comment]
Drawings on the `EmuCore` surface should be agnostic to the screen gap. I don't know if other cores' non-sync settings affect the buffer size, but this one certainly shouldn't.

**Comment by CasualPokePlayer** [comment]
The video buffer has the screen gap now, and you can see this with the buffer width/height changing. This is due to the changes just being all screen control stuff being done core side (like Citra) using melonDS upstream's helper code.

**Comment by YoshiRulz** [comment]
Yes, that's what I'm complaining about. While it wasn't working before unless the screen gap was 0, I still consider this a regression. Lua scripts should not have to draw at `buffer_height - 100` or whatever.

**Comment by CasualPokePlayer** [comment]
Layout, rotation, screen gap, screen invert, and screen gap will affect the emu core surface as the video buffer is directly affected. This is also the same as Citra. These cores have upstream helper code to handle screen controls, which have to be invoked in some capacity if OpenGL renderers are being used (and it's not hard to apply it to the software renderer in melon's case). So the screen control is handled core side, and that's what the frontend receives as is. If it's possible perhaps these points could just be transformed within the screen control class, somehow (not sure if that's fundamentally possible).

**Comment by YoshiRulz** [comment]
I still don't like this, though I'll concede that it has precedent in the "overscan" settings for the likes of the NES and Master System. Hence I'm re-opening this as a feature request.

**Comment by YoshiRulz** [comment]
@CasualPokePlayer If this is non-trivial to implement for melonDS or Encore, you can push it into the next milestone.

**Comment by CasualPokePlayer** [comment]
Is this just wanting the reverse of client.transformPoint?

**Comment by YoshiRulz** [comment]
No. The input co-ordinates should be relative to some arbitrary standard (in the case of NDS, `(0, 0)` through `(255, 383)` for the 2 displays stacked), and I think the output co-ordinates should be relative to the `EmuCore` surface. Then that can be passed on to `transformPoint` if necessary.
